### ISSUE TITLE
npm i lost dependencies

### ISSUE THREAD

**Issue by nuintun** [npm/npm#16902]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

### supporting information:

 - `npm -v` prints: 5.0.1
 - `node -v` prints: v8.0.0
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: Windows
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

[file-send](https://github.com/nuintun/file-send?ts=2) deps:
```json
"dependencies": {
  "ms": "^2.0.0",
  "etag": "^1.8.0",
  "fresh": "^0.5.0",
  "mime-types": "^2.1.15",
  "destroy": "^1.0.4",
  "colors": "^1.1.2",
  "micromatch": "^3.0.2",
  "encodeurl": "^1.0.1",
  "escape-html": "^1.0.3",
  "on-finished": "^2.3.0",
  "range-parser": "^1.2.0",
  "readable-stream": "^2.2.9"
}
```

[fengine](https://github.com/nuintun/fengine?ts=2) deps:
```json
"dependencies": {
  "colors": "^1.1.2",
  "file-send": "https://github.com/nuintun/file-send/archive/master.tar.gz",
  "js-yaml": "^3.8.4",
  "mime-types": "^2.1.15",
  "tilda": "^4.4.10"
}
```
when in project fengine I'm run ```npm i```

deps ```ms``` lost !!!

![qq 20170602113230](https://cloud.githubusercontent.com/assets/1730277/26709709/3c964c92-4787-11e7-8b91-eec9b58813ac.png)

**Comment by jasonburrows** [comment]
Yeah on my project npm install completes successfully, but npm list shows like 20 unmet dependencies that have no reason to be unmet (they simply were not installed).

For example, just installing the package 'onchange' (npm install onchange) will succeed, but if you then do an npm list there are 8 unmet dependencies for that package alone.

**Comment by zhu-meng-han** [comment]
The same problem。How to deal with it？

**Comment by jasonburrows** [comment]
@zmh7057 delete the package-lock.json file and it will get you past this for now.

**Comment by zkat** [comment]
This is fixed as of #17508.

You can use it right now using the canary (`npm i -g npmc@latest && npmc -v`). The patch will be available in mainline npm next Wednesday.
